bemani/backend/hellopopn/hellopopn.py

```diff
@@ -0,0 +1,267 @@
+        last.set_attribute('music_id', profile.get_str('music_id'))
+        last.set_attribute('style', profile.get_str('style'))
+
+        self.update_play_statistics(userid)
+        
+        return root
+
+    def handle_game_load_m_request(self, request: Node) -> Node:
+        #Load scores
+        userid = self.data.remote.user.from_refid(self.game, self.version, request.attribute('refid'))
+        scores = self.data.remote.music.get_scores(self.game, self.version, userid)
+
+        root = Node.void('game')
+        sortedscores: Dict[int, Dict[int, Score]] = {}
+        for score in scores:
+            if score.id not in sortedscores:
+                sortedscores[score.id] = {}
+            sortedscores[score.id][score.chart] = score
+        
+        for song in sortedscores:
+            for chart in sortedscores[song]:
+                score = sortedscores[song][chart]
+
+                music = Node.void('music')
+                root.add_child(music)
+                music.set_attribute('music_id', str(score.id))
+
+                style = Node.void('style')
+                music.add_child(style)
+                style.set_attribute('id', "0")
+
+                level = Node.void('level')
+                style.add_child(level)
+
+                level.set_attribute('id', str(score.chart))
+                level.set_attribute('score', str(score.points))
+                level.set_attribute('clear_type', str(score.data.get_int('clear_type')))
+
+        return root
+
+    def handle_game_save_request(self, request: Node) -> Node:
+        #Save profile data 
+        root = Node.void('game')
+
+        userid = self.data.remote.user.from_refid(self.game, self.version, request.attribute('refid'))
+        oldprofile = self.get_profile(userid)
+
+        newprofile = copy.deepcopy(oldprofile)
+
+        last = request.child('last')
+        newprofile.replace_str('chara', last.attribute('chara'))
+        newprofile.replace_str('level', last.attribute('level'))
+        newprofile.replace_str('music_id', last.attribute('music_id'))
+        newprofile.replace_str('style', last.attribute('style'))
+        newprofile.replace_str('love', last.attribute('love'))
+
+        self.put_profile(userid, newprofile)
+
+
+        game_config = self.get_game_config()
+        force_unlock_songs = game_config.get_bool("force_unlock_songs")
+        #if we were on force unlock mode, achievements will not be modified
+        if force_unlock_songs is False:
+          achievements = self.data.local.user.get_achievements(self.game, self.version, userid)
+          love = last.attribute('love')
+          for achievement in achievements:
+                  if achievement.type == 'toki_love' and achievement.id == int(last.attribute('chara')):
+                      love = str(int(achievement.data["love"]) + 1)
+                      if achievement.data["love"] == "5":
+                          love = "5"
```

Comment:
since server only sends what chara we level up, server side needs to plus 1, 5 is the max level, so if its more than 5 should stay as 5, ill try to make a cleaner form